packages/peregrine/src/Router/resolveUnknownRoute.js

```diff
@@ -107,6 +109,11 @@ function fetchRoute(opts) {
     })
         .then(res => res.json())
         .then(res => {
+            if(res.urlResolver == null && checkCount < 1) {
```

Comment:
It would be better to put this "retry" logic at the point of the invocation (ie. within `remotelyResolveRoute`) so we're not maintaining a global variable. You could do something similar to the following:

```js
let resolved = await fetchRoute(opts)
if (!resolved) {
  opts.route.replace('.html', '');
  resolved = await fetchRoute(opts)
}
return resolved;
```

I'm not actually sure if that would return something or throw... I leave it to you to figure out :P

Reply:
Cheers for the idea, i dropped the current way and added a async await function, with a check for html extension to make sure it doesnt infiante loop